### DISCUSSION TITLE
Serializer, how to deserialize BackedEnum? Cannot instantiate enum - solved

### DISCUSSION THREAD

**Discussion by xorock** [symfony/symfony#46972] [Q&A]
Hello everyone, 
I'm trying to deserialize XML document and everything works fine, except I don't know how to convert property to enum. I'm getting error "`<b>Fatal error</b>:  Uncaught Error: Cannot instantiate enum [...]\Response\ProductCategory in [...]/vendor/symfony/serializer/Normalizer/AbstractNormalizer.php:409`". My packages are in latest versions (6.1), PHP 8.1.

edit: sorry, I was looking at wrong class. Not it's fine.

```
class Product
{
    public function __construct(
        #[SerializedName('@id')]
        public readonly int $id,
        #[SerializedName('@cat')]
        public readonly ?ProductCategory $productCategory,
    )
    {
    }
}

enum ProductCategory: string
{
    case LD = 'LD';
    case STD = 'STD';
}

// invocation
        $classMetadataFactory = new ClassMetadataFactory(new AnnotationLoader(new AnnotationReader()));
        $metadataAwareNameConverter = new MetadataAwareNameConverter($classMetadataFactory);

        $xmlEncoder = new XmlEncoder([
            'xml_format_output' => true,
            'xml_encoding' => 'utf-8',
            'xml_version' => '1.0',
            'remove_empty_tags' => false,
        ]);

        $this->xmlEncoder = $xmlEncoder;

        $encoders = [$xmlEncoder];
        $normalizers = [
            new DateTimeNormalizer([
                DateTimeNormalizer::FORMAT_KEY => DateTimeInterface::RFC3339_EXTENDED,
                DateTimeNormalizer::TIMEZONE_KEY => null,
            ]),
            new ArrayDenormalizer(),
            new BackedEnumNormalizer(),
            new ObjectNormalizer($classMetadataFactory, $metadataAwareNameConverter, null, new PhpDocExtractor()),
        ];
        $this->serializer = new Serializer($normalizers, $encoders);
```

**Comment by stof** [comment]
> edit: sorry, I was looking at wrong class. Not it's fine.

What do you mean with this edit ? That you don't actually have an issue anymore ?

**Comment by xorock** [comment]
Hi, yes. It was my fault. I modified the wrong class. 
The only comment I have here is that in the documentation there is not even a word about the BakedEnumNormalizer and only by searching the web can you find out about it. It would be nice to add some comment about it to the docs.

**Reply by ToshY** [reply]
Had the same issue but I've found `BackedEnumNormalizer`in the recent docs under [Build-in Normalizers](https://symfony.com/doc/current/components/serializer.html#built-in-normalizers).
